src/processors/modulation/scanner_vibrato/ScannerVibrato.cpp

```diff
@@ -158,16 +182,45 @@ void ScannerVibrato::processAudio (AudioBuffer<float>& buffer)
                  ch,
                  numSamples,
                  tapsOutData,
-                 inData = audioInBuffer.getReadPointer (ch)] (auto mode)
+                 inData = audioInBuffer.getReadPointer (ch % numInChannels)] (auto mode)
                 {
                     constexpr Mode mode_c = mode;
                     wdf[ch].processBlock<mode_c> (inData, tapsOutData, numSamples);
                 },
                 static_cast<Mode> (modeIndex));
 
             auto* outData = audioOutBuffer.getWritePointer (ch);
-            for (int i = 0; i < ScannerVibratoWDF::numTaps; ++i)
-                FloatVectorOperations::addWithMultiply (outData, tapsOutData[i], modMixData[i], numSamples);
+            if (stereoMode)
+            {
+                if (ch == 1)
+                {
+                    // invert right phase
+                    FloatVectorOperations::negate (modMixDataCh1[0], modMixDataCh1[0], numSamples);
+                    FloatVectorOperations::add (modMixDataCh1[0], 1.0f, numSamples);
+
+                    // process right and add to output
+                    for (int i = ScannerVibratoWDF::numTaps - 1; i >= 0; --i)
+                    {
+                        tapMixTable[i].process (modMixDataCh1[0], modMixDataCh1[i], numSamples);
+                        FloatVectorOperations::addWithMultiply (outData, tapsOutData[i], modMixDataCh1[i], numSamples);
+                    }
+                }
+
+                else
+                {
+                    // process left and add to output
+                    for (int i = ScannerVibratoWDF::numTaps - 1; i >= 0; --i)
+                    {
+                        tapMixTable[i].process (modMixDataCh0[0], modMixDataCh0[i], numSamples);
+                        FloatVectorOperations::addWithMultiply (outData, tapsOutData[i], modMixDataCh0[i], numSamples);
+                    }
+                }
+            }
+            else
+            {
+                for (int i = ScannerVibratoWDF::numTaps - 1; i >= 0; --i)
+                    FloatVectorOperations::addWithMultiply (outData, tapsOutData[i], modMixDataCh0[i], numSamples);
+            }
```

Comment:
This is a pretty long logic statement that could be narrowed down further? do you have any ideas on the best way to cut it down?